### DISCUSSION TITLE
Epsilon 2.5 Released

### DISCUSSION THREAD

**Discussion by kolovos** [eclipse-epsilon/epsilon#98] [Announcements]
Epsilon 2.5 was released on April 15, 2024.

# New and Noteworthy

- Added support for JSON, YAML and CDO models
- Improved performance of ECL's match trace
- Added Ant tasks for Pinset and EMG
- Added support for [partial enumeration literals](https://eclipse.dev/epsilon/doc/eol/#enumerations) to EOL
- Added support for varargs to EOL reflection
- Updated version of PlantUML used in Picto
- Added support for specifying EGL formatters from EGX
- All languages now reuse imports
- Added support for Edge in Picto
- Added support to Flexmi for importing/referencing non-Flexmi (e.g. XMI) resources
- Performance improvements in EGL (protected regions)

**Comment by Haochi1222** [comment]
Hi ,when I install Epsilon2.5,it comes error that ,
An error occurred while collecting items to be installed
session context was:(profile=DefaultProfile, phase=org.eclipse.equinox.internal.p2.engine.phases.Collect, operand=, action=).
Result of processing steps.
OK
OK
Public key not found for 6640597757109451687.
OK
Result of processing steps.
OK
OK
Public key not found for 6640597757109451687.
OK

**Reply by Haochi1222** [reply]
I have tried 
http://download.eclipse.org/epsilon/updates/2.5/
&https://www.eclipse.org/downloads/download.php?file=/epsilon/updates/2.5/epsilon-2.5-site.zip
Both of them don't work.

**Reply by kolovos** [reply]
Could you please try to install a clean instance of Eclipse and Epsilon through the Eclipse Installer as shown at the top of https://eclipse.dev/epsilon/download/ and let me know if this works for you?

**Reply by Haochi1222** [reply]
yeah,I can install through Eclipse Installer. But for some reason, I need to use json2model in Sirius, so I find that it can't install in Sirius.

**Reply by kolovos** [reply]
You'll need to install Sirius manually through the `Help -> Install New Software` menu and the `https://download.eclipse.org/releases/2023-12` update site as shown below.

![image](https://github.com/eclipse/epsilon/assets/3584136/14cbf635-55ba-4e36-858f-d8acd4f4212b)

**Reply by Haochi1222** [reply]
I try this but 
![1720003166218](https://github.com/eclipse/epsilon/assets/59555272/9bfbb0e6-7ba3-4c35-be7f-473dc14985aa)
Your original request has been modified.
  "Sirius Properties Views - Specifier Support" is already installed, so an update will be performed instead.
  "Sirius Samples" is already installed, so an update will be performed instead.
  "Sirius Properties Views - Runtime Support" is already installed, so an update will be performed instead.
Cannot complete the install because of a conflicting dependency.
  Software being installed: Sirius Samples 7.3.0.202311270829 (org.eclipse.sirius.samples.feature.group 7.3.0.202311270829)
  Software currently installed: Obeo Designer Community 11.8.0.202303291536 (fr.obeo.dsl.designer.ce.product 11.8.0.202303291536)
  Only one of the following can be installed at once: 
    Sirius Family Example 7.3.0.202311270829 (org.eclipse.sirius.samples.family 7.3.0.202311270829)
    Sirius Family Example 7.1.0.202303020853 (org.eclipse.sirius.samples.family 7.1.0.202303020853)
  Cannot satisfy dependency:
    From: Obeo Designer Community 11.8.0.202303291536 (fr.obeo.dsl.designer.ce.product 11.8.0.202303291536)
    To: org.eclipse.equinox.p2.iu; org.eclipse.sirius.samples.feature.group [7.1.0.202303020853,7.1.0.202303020853]
  Cannot satisfy dependency:
    From: Sirius Samples 7.1.0.202303020853 (org.eclipse.sirius.samples.feature.group 7.1.0.202303020853)
    To: org.eclipse.equinox.p2.iu; org.eclipse.sirius.samples.family [7.1.0.202303020853,7.1.0.202303020853]
  Cannot satisfy dependency:
    From: Sirius Samples 7.3.0.202311270829 (org.eclipse.sirius.samples.feature.group 7.3.0.202311270829)
    To: org.eclipse.equinox.p2.iu; org.eclipse.sirius.samples.family [7.3.0.202311270829,7.3.0.202311270829]

**Reply by Haochi1222** [reply]
it seems I can't install Obeo Designer Community 11.8?

**Reply by kolovos** [reply]
I'm not familiar with Obeo Designer Community I'm afraid.

**Comment by kolovos** [comment]
@Haochi1222 I've been able to install Epsilon 2.5 on top of Obeo Designer Community using the following steps:

- Start with a clean instance of Obeo Designer Community 11.8
- Install `XSD - XML Schma Definition SDK` from https://download.eclipse.org/modeling/emf/emf/builds/release/2.26
- In the `Preferences` of Obeo Designer Community, go to `Install/Update -> Available Software Sites` and untick the `Eclipse Repository` item
- Install Epsilon from http://download.eclipse.org/epsilon/updates/2.5/ excluding the following features
    - `Epsilon Sirius Integration`
    - `Epsilon UML Integration`
    - `Epsilon CDO Integration`

**Reply by Haochi1222** [reply]
![image](https://github.com/eclipse/epsilon/assets/59555272/3fc37936-37d0-4cbd-b4c6-98222afacc37)
![屏幕截图 2024-07-04 152654](https://github.com/eclipse/epsilon/assets/59555272/aa3bed80-bb13-411a-82b0-8fdce11cf2da)
Thank you so much for your detailed response to my question! But when I try the 3rd step, it comes the same error as before , now I'm trying to use both epsilon and obeo Designer to work. I sincerely appreciate the time and effort you took to help me.

**Reply by kolovos** [reply]
You're right. I've updated the [instructions above](https://github.com/eclipse/epsilon/discussions/98#discussioncomment-9952006) and it should work now.

**Reply by Haochi1222** [reply]
wowww！That‘s amazing， I successfully install it！This has been a huge help to me. Thank you for your invaluable assistance!
![1e05a102da59895e066d1833d73c715](https://github.com/eclipse/epsilon/assets/59555272/f96f84fd-5f4b-4421-845e-7ddfa027593f)

**Reply by kolovos** [reply]
You're welcome!

**Comment by kolovos** [comment]
Reopening as announcements are not meant to be closed.
